assignments/prototype-refactor.js

```diff
@@ -7,3 +7,222 @@ Prototype Refactor
+
+  At the bottom of this file are 3 objects that all end up inheriting from Humanoid. 
+  
+  Use the objects at the bottom of the page to test your constructor functions.
+  
+  Each constructor function has unique properites and methods that are defined in their block comments below:
+*/
+
+/*
+  === GameObject ===
+  * createdAt
+  * dimensions
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+*/
+
+class GameObject {
+  constructor(attributes) {
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+  }
+  destroy() {
+    return `${this.name} was removed from the game.`;
+  }
+}
+
+/*
+  === CharacterStats ===
+  * hp
+  * name
+  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+  * should inherit destroy() from GameObject's prototype
+*/
+class CharacterStats extends GameObject {
+  constructor(stats) {
+    super(stats);
+    this.name = stats.name;
+    this.hp = stats.hp;
+    this.alive = stats.alive;
+  }
+  takeDamage() {
+    return `${this.name} took damage`;
+  }
+}
+
+/*
+  === Humanoid ===
+  * faction
+  * weapons
+  * language
+  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+  * should inherit destroy() from GameObject through CharacterStats
+  * should inherit takeDamage() from CharacterStats
+*/
+
+class Humanoid extends CharacterStats {
+  constructor(data) {
+    super(data);
+    this.faction = data.faction;
+    this.weapons = data.weapons.toString();
+    this.language = data.language;
+  }
+  greet() {
+    return `${this.name} offers a greeting in ${this.language}.`;
+  }
+}
+
+/*
+  * Inheritance chain: GameObject -> CharacterStats -> Humanoid
+  * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
+  * Instances of CharacterStats should have all of the same properties as GameObject.
+*/
+
+// Stretch task:
+// * Create Villian and Hero constructor functions that inherit from the Humanoid constructor function.
+// * Give the Hero and Villians different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
+// * Create two new objects, one a villian and one a hero and fight it out with methods!
+
+class Villian extends Humanoid {
+  constructor(info) {
+    super(info);
+  }
+  fireball(character) {
+    if (character.hp > 1) {
+      character.hp--;
+      return `${this.name} casts fireball. ${character.takeDamage()}. ${
+        character.name
+      } now has ${character.hp} hp.`;
+    } else {
+      character.hp--;
+      return `${this.name} casts fireball. ${character.takeDamage()}. ${
+        character.name
+      } now has ${character.hp} hp. ${character.destroy()}`;
+    }
+  }
+}
+
+class Hero extends Humanoid {
+  constructor(info) {
+    super(info);
+  }
+  smite(character) {
```

Comment:
Good job w/ your hero and villain methods. Very creative and you did it in a super clean way (easy to read, no wasted space, invoking methods in your string literals.